multiformats_cid/cid.py

```diff
@@ -270,7 +281,7 @@ def from_bytes(cidbytes):
         if len(cid) < 2:
             raise ValueError('cid length is invalid')
 
-        data = cid[1:]
+        data = bytes(cid[1:])  # This is a bit weird...
```

Comment:
This is one place that I am not super happy about, the cid is declared just above as 

```py
cid = multibase.decode(cidbytes)
```

and the decode method signature returns string. However, just a few lines below we pass this "data" into two places

```py
codec = multicodec.get_codec(data)
multihash = multicodec.remove_prefix(data)
```

Both which only accept bytes!

I did notice that `mypy` does not complain about this with or without the change made here (it is only my IDE that indicates something is wrong here).

If you are not comfortable with this change it can be reverted by applying this suggestion:

```suggestion
        data = cid[1:]
```

Reply:
That's a valid concern yeah. I prefer to keep the bytes casting you added as it makes it clear what the type of *data* is.